hearth/lib/hearth/http/client.rb

```diff
@@ -81,28 +82,56 @@ def initialize(options = {})
       # @param [Response] response
       # @return [Response]
       def transmit(request:, response:, **options)
-        http = create_http(request.uri)
-        http.set_debug_output(options[:logger]) if @debug_output
-        configure_timeouts(http)
-
-        if request.uri.scheme == 'https'
-          configure_ssl(http)
-        else
-          http.use_ssl = false
+        # http = create_http(request.uri)
+        # http.set_debug_output(options[:logger]) if @debug_output
+        # configure_timeouts(http)
+        #
+        # if request.uri.scheme == 'https'
+        #   configure_ssl(http)
+        # else
+        #   http.use_ssl = false
+        # end
+        endpoint = request.uri
+        pool = ConnectionPool.for(config)
+        session = pool.session_for(endpoint) do
+          start_session(endpoint)
         end
+        _transmit2(session, request, response)
+        pool.offer(endpoint, session)
```

Comment:
Just confirming: this isn't done in a finally because we only `release` a connection when its successful right?

Reply:
We don't want to release/finish a connection when it's successful. It goes back into the pool. An ensure block would execute in any case.

Reply:
I've re-arranged things to make this easier to understand hopefully